src/widgetastic_patternfly4/duallistselector.py

```diff
@@ -0,0 +1,132 @@
+    SELECTED_ITEMS = (
+        ".//li[@aria-selected='true' and contains(@class, 'dual-list-selector__list-item')]"
+    )
+    move_all_right = Button(locator=".//button[@aria-label='Add all']")
+    move_selected_right = Button(locator=".//button[@aria-label='Add selected']")
+    move_selected_left = Button(locator=".//button[@aria-label='Remove selected']")
+    move_all_left = Button(locator=".//button[@aria-label='Remove all']")
+
+    @property
+    def _left_list(self):
+        return self.browser.elements(self.LIST_ITEMS, parent=self)[0]
+
+    @property
+    def _right_list(self):
+        return self.browser.elements(self.LIST_ITEMS, parent=self)[-1]
+
+    @property
+    def _left_title(self):
+        return self.browser.elements(self.SECTION_TITLE, parent=self)[0].text
+
+    @property
+    def _right_title(self):
+        return self.browser.elements(self.SECTION_TITLE, parent=self)[-1].text
+
+    @property
+    def _left_elements(self):
+        return self.browser.elements(self.ITEMS, parent=self._left_list)
+
+    @property
+    def _right_elements(self):
+        return self.browser.elements(self.ITEMS, parent=self._right_list)
+
+    @property
+    def _selected_left_elements(self):
+        return self.browser.elements(self.SELECTED_ITEMS, parent=self._left_list)
+
+    @property
+    def _selected_right_elements(self):
+        return self.browser.elements(self.SELECTED_ITEMS, parent=self._right_list)
+
+    def move_all_items_right(self):
+        self.move_all_right.click()
+
+    def move_all_items_left(self):
+        self.move_all_left.click()
+
+    def move_selected_items_right(self):
+        self.move_selected_right.click()
+
+    def move_selected_items_left(self):
+        self.move_selected_left.click()
+
+    def read(self, selected_only=False):
+        """Read items on left and right sides, if selected_only = True, read only selected items"""
+        data = dict()
+
+        if selected_only:
+            right_elements = self._selected_right_elements
+            left_elements = self._selected_left_elements
+        else:
+            right_elements = self._right_elements
+            left_elements = self._left_elements
+
+        data[self._left_title] = [el.text for el in left_elements]
+        data[self._right_title] = [el.text for el in right_elements]
+        return data
+
+    def reset_selected(self, left_items=True):
+        if left_items:
+            _elements = self._left_elements
+        else:
+            _elements = self._right_elements
+
+        for element in _elements:
+            if element.get_attribute("aria-selected") == "true":
+                element.click()
+
+    def select(self, items, left_items=True):
+        """Select certain items based on text. Defaults to select left_items"""
+        if left_items:
+            _elements = self._left_elements
+        else:
+            _elements = self._right_elements
+
+        _text_items = [el.text for el in _elements]
```

Comment:
Lets use `self.browser.text(el)`. for wt better go with browser class than direct web-element interaction.

Reply:
didn't know about it, fixed! TIL